src/modules/Dropdown/DropdownItem.js

```diff
@@ -11,44 +12,85 @@ import {
 } from '../../lib'
 import { createIcon, createShorthand } from '../../factories'
 
-function DropdownItem(props) {
-  const {
-    active,
-    children,
-    className,
-    disabled,
-    description,
-    icon,
-    onClick,
-    selected,
-    text,
-    value,
-  } = props
-
-  const handleClick = (e) => {
-    if (onClick) onClick(e, value)
+export default class DropdownItem extends Component {
+  constructor() {
+    super()
+
+    this.state = {
+      hovered: false,
+    }
   }
 
-  const classes = cx(
-    useKeyOnly(active, 'active'),
-    useKeyOnly(disabled, 'disabled'),
-    useKeyOnly(selected, 'selected'),
-    'item',
-    className,
-  )
-  // add default dropdown icon if item contains another menu
-  const iconName = icon || childrenUtils.someByType(children, 'DropdownMenu') && 'dropdown'
-  const ElementType = getElementType(DropdownItem, props)
-  const rest = getUnhandledProps(DropdownItem, props)
-
-  return (
-    <ElementType {...rest} className={classes} onClick={handleClick}>
-      {createShorthand('span', val => ({ className: 'description', children: val }), description)}
-      {createIcon(iconName)}
-      {text}
-      {children}
-    </ElementType>
-  )
+  render() {
+    const {
+      active,
+      children,
+      className,
+      disabled,
+      description,
+      icon,
+      onClick,
+      onMouseEnter,
+      onMouseLeave,
+      selected,
+      text,
+      value,
+    } = this.props
+
+    const handleClick = (e) => {
+      if (onClick) onClick(e, value)
+    }
+
+    const handleMouseEnter = (e) => {
+      if (onMouseEnter) onMouseEnter(e, value)
+
+      this.setState({ hovered: true })
+    }
+
+    const handleMouseLeave = (e) => {
+      if (onMouseLeave) onMouseLeave(e, value)
+
+      this.setState({ hovered: false })
+    }
+
+    const classes = cx(
+      useKeyOnly(active, 'active'),
+      useKeyOnly(disabled, 'disabled'),
+      useKeyOnly(selected, 'selected'),
+      'item',
+      className,
+    )
+    // add default dropdown icon if item contains another menu
+    const iconName = icon || childrenUtils.someByType(children, 'DropdownMenu') && 'dropdown'
+    const ElementType = getElementType(DropdownItem, this.props)
+    const rest = getUnhandledProps(DropdownItem, this.props)
+
+    let newChildren = children
+
+    if (Array.isArray(children)) {
+      newChildren = Children.map(children, (child) => {
+        if (child.type !== DropdownMenu) {
+          return child
+        }
+
+        return cloneElement(child, { opened: this.state.hovered })
```

Comment:
We're trying to get away from looping children.  Why? If a user passes their own menu component, this will fail since it will not be a `DropdownMenu`.  The same is true if they decorate the Menu, wrapping it in a HOC.  Also if it is augmented with the `as` prop.

Because of this, I don't want to add any features that deal with parsing children since they almost always cause bugs.

Let's remove this, which means we can remove the state `hovered`, and also allows us to keep this as a functional component.  The parent / implementing component can manage the state.  The mouse props will be spread in `rest` as well.  Assuming I haven't overlooked something, I don't believe there is a need to extend the MenuItem to accomplish this feature.